railties/test/application/rake/dev_test.rb

```diff
@@ -17,33 +17,47 @@ def teardown
 
       test "dev:cache creates file and outputs message" do
         Dir.chdir(app_path) do
-          output = rails("dev:cache")
-          assert File.exist?("tmp/caching-dev.txt")
-          assert_match(/Development mode is now being cached/, output)
+          stderr = capture(:stderr) do
+            output = run_rake_dev_cache
+            assert File.exist?("tmp/caching-dev.txt")
+            assert_match(/Development mode is now being cached/, output)
+          end
+          assert_match(/DEPRECATION WARNING: Using `bin\/rake dev:cache` is deprecated and will be removed in Rails 6.1/, stderr)
         end
       end
 
       test "dev:cache deletes file and outputs message" do
         Dir.chdir(app_path) do
-          rails "dev:cache" # Create caching file.
-          output = rails("dev:cache") # Delete caching file.
-          assert_not File.exist?("tmp/caching-dev.txt")
-          assert_match(/Development mode is no longer being cached/, output)
+          stderr = capture(:stderr) do
+            run_rake_dev_cache # Create caching file.
+            output = run_rake_dev_cache # Delete caching file.
+            assert_not File.exist?("tmp/caching-dev.txt")
+            assert_match(/Development mode is no longer being cached/, output)
+          end
+          assert_match(/DEPRECATION WARNING: Using `bin\/rake dev:cache` is deprecated and will be removed in Rails 6.1/, stderr)
         end
       end
 
       test "dev:cache touches tmp/restart.txt" do
         Dir.chdir(app_path) do
-          rails "dev:cache"
-          assert File.exist?("tmp/restart.txt")
-
-          prev_mtime = File.mtime("tmp/restart.txt")
-          sleep(1)
-          rails "dev:cache"
-          curr_mtime = File.mtime("tmp/restart.txt")
-          assert_not_equal prev_mtime, curr_mtime
+          stderr = capture(:stderr) do
+            run_rake_dev_cache
+            assert File.exist?("tmp/restart.txt")
+
+            prev_mtime = File.mtime("tmp/restart.txt")
+            sleep(1)
```

Comment:
Do we still have to `sleep` here since the Time are going to be compared (including milliseconds)

Reply:
I didn't touch this test since it was the rake task test that was already present and will be deleted in 6.1. But I don't think that sleep is necessary to start with. I'll get rid of it here too.